docs/INSTALL.md

```diff
@@ -3,69 +3,106 @@ Bhima Installation Guide
 
 ###### Getting the source
-Clone the source using git from the [bhima github repository] (https://github.com/IMA-WorldHealth/bhima-2.x).
+Clone the source using git from the [github repository](https://github.com/IMA-WorldHealth/bhima-2.x).
 
 ```bash
 git clone https://github.com/IMA-WorldHealth/bhima-2.X.git bhima-2.X
 cd bhima-2.X
 ```
 
 ###### Building from source
-We use the [gulpjs](http://www.gulpjs.com) build tool to build from source, installed with `npm`.
+All our build scripts are found the `package.json` file.  We use [gulpjs](http://www.gulpjs.com)
+internally, but you shouldn't ever need to call gulp explicitly.
+
+To execute the build scripts, you can use either `yarn` or `npm`.  We'll use
+`yarn` for the remainder of this guide.  Note that using `npm` may require you to
+use `npm run` where it says `yarn` below.
 
 ```bash
 # Inside the bhima-2.X/ directory
 
 # install all node modules
-npm install --no-progress --loglevel http
+yarn install
 
+# bower is now installed in ./node_modules/.bin/bower
 # install client-side dependencies with bower
 ./node_modules/.bin/bower install -f
+```
+
+The dependencies should now be set!
+
+BHIMA uses environmental variables to connect to the database and toggle features.
+These are found in the `.env.development` file included in the top level of the
+repository.  By default, the build script will copy all files named `.env.*` into
+the build folder `bin/` when building the application.  At runtime, the file
+corresponding to `.env.$NODE_ENV` will be used to configure the application.  For
+the default node instance, `NODE_ENV="development"`.  Please set this globally,
+if it is not set by default on your machine.
 
+Before building, edit your `.env.development` to set up your database connection
+parameters.  Their variables should be self-explanatory. Then, build the app with
+```bash
 # build the application
-npm run build
+NODE_ENV="development" yarn build
 ```
 
 ###### Creating a database
-The database structure is contained in the following files:
- 1. `server/models/*.sql`
-
-For an initial setup, bhima also includes a file `test/data.sql` that contains sample data to get the application up and running rapidly.  Build all three and customize further from within the running application.  Note that the database must be running SQL strict mode for all the logic to behave correctly.
-
-Alternative, you might use the `build-database.sh` script, customized with your environmental variables as shown below:
+_NOTE: BHIMA runs in `sql_mode='STRICT_ALL_TABLES'`.  While it is not necessary
+to have this set to build the database, the test will not pass unless the
+correct SQL_MODE is set._
+
+The database structure is contained in the `server/models/*.sql` files.  You can
+execute these one by one in the order below, or simply run `yarn build:db`.
+ 1. `server/models/schema.sql`
+ 2. `server/models/.sql`
```

Comment:
Is this file `server/models/.sql` exists ?

Reply:
Ha!  Good catch :)